src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -300,15 +301,19 @@ void StorageObjectStorageQueue::shutdown(bool is_drop)
             tryLogCurrentException(log);
         }
 
-        files_metadata->shutdown();
+        ObjectStorageQueueMetadataFactory::instance().remove(zk_path, getStorageID());
         files_metadata.reset();
     }
     LOG_TRACE(log, "Shut down storage");
 }
 
 void StorageObjectStorageQueue::drop()
 {
-    ObjectStorageQueueMetadataFactory::instance().remove(zk_path, getStorageID());
+    if (files_metadata)
```

Comment:
I think we do not need this code of `drop` now at all, because shutdown always goes before `drop`, so let's remove it?